addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +221,44 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if(!this.threadCache) {
```

Comment:
only happens when viewer not linked to a thread. `thread.cache()` always return an obj.

Reply:
Yes I know, what's your point ?

Reply:
Don't check relation with thread cache if that's basically equivalent to checking relation with thread.